modules/lib/lib-node/src/main/resources/lib/xp/node.ts

```diff
@@ -124,22 +126,22 @@ export interface SuggestionResult {
     }[];
 }
 
-export interface NodeQueryResult {
+export interface NodeQueryResult<AggregationOutput extends Record<string, AggregationsResult>> {
     total: number;
     count: number;
     hits: NodeQueryResultHit[];
-    aggregations?: Record<string, AggregationsResult>;
+    aggregations?: AggregationOutput;
```

Comment:
Can `aggregations` be `undefined` (not set)? I think that either `aggregations` would have a value. Or it would be `never`.

```suggestion
    aggregations: AggregationOutput;
```

Reply:
Yes, you are right. It will be either defined, or calculated to `never`.